File: onefile/makeit.rb
require 'json'

class Broadcast < Struct.new(:created_at, :events, :script_prompt, :script, :audio_file, :error)
  FILE_PATH = 'broadcast.json'

  def initialize(created_at: '', events: [], script_prompt: '', script: '', audio_file: '', error: '')
    super(created_at, events, script_prompt, script, audio_file, error)
  end
  
  def self.load
    return new unless File.exist?(FILE_PATH)
    
    data = JSON.parse(File.read(FILE_PATH))
    new(**data.transform_keys(&:to_sym))
  rescue JSON::ParserError
    new
  end
  
  def save
    File.write(FILE_PATH, JSON.pretty_generate(to_h))
    self
  end
  
  def to_h
    super.transform_keys(&:to_s)
  end
end

File: onefile/makeit.py
import json
from dataclasses import dataclass, field
from typing import List

@dataclass
class Record:
    created_at: str = ""
    events: List[str] = field(default_factory=list)
    script_prompt: str = ""
    script: str = ""
    audio_file: str = ""
    error: str = ""
    
    FILE_PATH = "data.json"
    
    @classmethod
    def load(cls):
        try:
            with open(cls.FILE_PATH, 'r') as f:
                data = json.load(f)
                return cls(**data)
        except (FileNotFoundError, json.JSONDecodeError):
            return cls()
    
    def save(self):
        with open(self.FILE_PATH, 'w') as f:
            json.dump(self.__dict__, f, indent=2)
        return self